tests/compat.sh

```diff
@@ -217,7 +217,7 @@ filter_ciphersuites()
     # supports from the s_server help. (The s_client help isn't
     # accurate as of 1.0.2g: it supports DTLS 1.2 but doesn't list it.
     # But the s_server help seems to be accurate.)
-    if ! $OPENSSL_CMD s_server -help 2>&1 | grep -q "^ *-$MODE "; then
+    if ! $OPENSSL_CMD s_server -help 2>&1 | grep -q "^ *-$O_MODE "; then
         M_CIPHERS=""
```

Comment:
Why would we skip m-m tests based on the availability of openssl features? This looks like a bug which affects test runs with `OPENSSL_LEGACY`. This bug was introduced in https://github.com/ARMmbed/mbedtls/pull/2779/commits/6eece5b666a36094f5ebef93b9683df06b81c698.

Reply:
For example https://ci.trustedfirmware.org/job/mbed-tls-pr-head/job/PR-5630-head/1/execution/node/1276/log/ lacks `m->m` and `m->G` tests under `test: compat.sh DES, 3DES & NULL (full config)`.